lib/cancan/ability/strong_parameter_support.rb

```diff
@@ -0,0 +1,25 @@
+module CanCan
+  module Ability
+    module StrongParameterSupport
+      # Return an array of attributes suitable for use with strong parameters
+      def permitted_attributes(action, subject)
+        @permitted_attributes ||= {}
+        @permitted_attributes[[action, subject]] ||= allowed_attributes(action, subject)
+      end
+
+      private
+
+      def allowed_attributes(action, subject)
+        attributes = relevant_rules(action, subject).flat_map do |rule|
+          if rule.attributes.empty? && subject.class == Class # empty attributes is an 'all'
+            subject.instance_methods.map(&:to_sym)
```

Comment:
you are actually taking all methods and not just attributes here, or I got it wrong? in that case it smells a little bit, what do you think?

Reply:
You're absolutely right. I should tighten it up.